api/src/pcapi/core/educational/api/offer.py

```diff
@@ -556,36 +553,18 @@ def edit_collective_offer_public(
             offer.collectiveStock.bookingLimitDatetime = new_values.get(
                 "beginningDatetime", offer.collectiveStock.beginningDatetime
             )
-            if collective_stock_unique_booking:
-                collective_stock_unique_booking.confirmationLimitDate = value
-        elif key == "beginningDatetime":
-            offer.collectiveStock.beginningDatetime = value
-            if collective_stock_unique_booking:
-                update_collective_booking_cancellation_limit_date(collective_stock_unique_booking, value)
-                update_collective_booking_educational_year_id(collective_stock_unique_booking, value)
-        elif key == "price":
-            offer.collectiveStock.price = value
-            if collective_stock_unique_booking:
-                collective_stock_unique_booking.amount = value
         elif key in stock_fields:
             setattr(offer.collectiveStock, key, value)
         elif key in offer_fields:
             setattr(offer, key, value)
         else:
             raise ValueError(f"unknown field {key}")
 
-    # if the booking limit date is set in the future and a PENDING booking was automatically cancelled, set it back to PENDING
-    booking_limit_value = offer.collectiveStock.bookingLimitDatetime
-    if (
-        feature.FeatureToggle.ENABLE_COLLECTIVE_NEW_STATUSES.is_active()
-        and "bookingLimitDatetime" in new_values
-        and booking_limit_value
-        and booking_limit_value > datetime.datetime.utcnow()
-    ):
-        booking = offer.collectiveStock.lastBooking
-        if booking and booking.is_expired:
-            update_collective_booking_pending(booking)
-            db.session.add(booking)
+    api_shared.update_collective_stock_booking(
+        stock=offer.collectiveStock,
+        current_booking=collective_stock_unique_booking,
+        beginning_has_changed="beginningDatetime" in new_values,
```

Comment:
Mettre l'attribut en entier ici n'est pas un luxe pour la lisibilité

```suggestion
        beginning_datetime_has_changed="beginningDatetime" in new_values,
```